stan/math/prim/scal/meta/is_constant_struct.hpp

```diff
@@ -2,18 +2,27 @@
 #define STAN_MATH_PRIM_SCAL_META_IS_CONSTANT_STRUCT_HPP
 
 #include <stan/math/prim/scal/meta/is_constant.hpp>
+#include <stan/math/prim/scal/meta/conjunction.hpp>
 
 namespace stan {
-
 /**
- * Metaprogram to determine if a type has a base scalar
- * type that can be assigned to type double.
- * @tparam T Types to test
+ * Defines a public enum named value and sets it to true(1)
+ * if the provided type is constant, false(0) otherwise.
+ * This helper struct is used in the
+ * is_constant_struct metaprogram.
+ * @tparam type to test
  */
 template <typename T>
-struct is_constant_struct {
+struct is_constant_struct_helper {
   enum { value = is_constant<T>::value };
 };
+/**
+ * Metaprogram to determine if a type has a base scalar
+ * type that can be assigned to type double.
```

Comment:
How about:

```
Metaprogram defining an enum `value` which is `true` if all of 
the type parameters are constant (i.e., primtive types) 
and `false` otherwise.
```

Or you can point back to the doc for the helper function for the notion of constant.  It's nice to have only one source of doc, but this is so short I'd just include it redundantly here.